Loop/View Controllers/StatusTableViewController.swift

```diff
@@ -2014,56 +2034,185 @@ extension StatusTableViewController: SettingsViewModelDelegate {
+            case .createdAndOnboarded:
+                log.default("Service with identifier '%{public}@' created and onboarded", identifier)
+            }
+        }
+    }
+
+    func gotoService(withIdentifier identifier: String) {
+        guard let serviceUI = deviceManager.servicesManager.activeServices.first(where: { $0.serviceIdentifier == identifier }) as? ServiceUI else {
+            return
+        }
+        showServiceSettings(serviceUI)
+    }
+
+    fileprivate func showServiceSettings(_ serviceUI: ServiceUI) {
+        var settingsViewController = serviceUI.settingsViewController(colorPalette: .default)
+        settingsViewController.serviceOnboardDelegate = self
+        settingsViewController.completionDelegate = self
+        show(settingsViewController, sender: self)
     }
 }
 
-extension StatusTableViewController: ServiceSettingsDelegate {
-    func serviceSettingsNotifying(_ object: ServiceSettingsNotifying, didDeleteService service: Service) {
-        deviceManager.servicesManager.removeActiveService(service)
+extension StatusTableViewController: ServiceCreateDelegate {
+    func serviceCreateNotifying(didCreateService service: Service) {
+        log.default("Service with identifier '%{public}@' created", service.serviceIdentifier)
+        deviceManager.servicesManager.addActiveService(service)
     }
 }
 
-extension StatusTableViewController: ServicesViewModelDelegate {
-    func addService(identifier: String) {
-        setupService(withIdentifier: identifier)
+extension StatusTableViewController: ServiceOnboardDelegate {
+    func serviceOnboardNotifying(didOnboardService service: Service) {
+        precondition(service.isOnboarded)
+        log.default("Service with identifier '%{public}@' onboarded", service.serviceIdentifier)
     }
-    func gotoService(identifier: String) {
-        guard let serviceUI = deviceManager.servicesManager.activeServices.first(where: { $0.serviceIdentifier == identifier }) as? ServiceUI else {
+}
+
+// MARK: - Onboarding
+
+extension StatusTableViewController {
+    fileprivate func setupOnboarding(withIdentifier identifier: String) {
+        guard let onboardingUIType = deviceManager.pluginManager.getOnboardingTypeByIdentifier(identifier) else {
             return
         }
-        showServiceSettings(serviceUI)
+
+        let onboarding = onboardingUIType.createOnboarding()
+        var onboardingViewController = onboarding.onboardingViewController(preferredGlucoseUnit: deviceManager.preferredGlucoseUnit,
+                                                                           cgmManagerProvider: self,
+                                                                           pumpManagerProvider: self,
+                                                                           serviceProvider: self,
+                                                                           colorPalette: .default)
+        onboardingViewController.onboardingDelegate = self
```

Comment:
Whoof.  This kind of code always seems like a bad smell...i.e. this class is doing too much.  (not suggesting making changes, just noting it here...)